src/jit/importer.cpp

```diff
@@ -1176,11 +1176,11 @@ GenTree* Compiler::impAssignStructPtr(GenTree*             destAddr,
                     // We change this to a GT_LCL_FLD (from a GT_ADDR of a GT_LCL_VAR)
                     lcl->ChangeOper(GT_LCL_FLD);
                     fgLclFldAssign(lcl->gtLclVarCommon.gtLclNum);
+                    lcl->gtType = src->gtType;
+                    asgType     = src->gtType;
                 }
 
-                lcl->gtType = src->gtType;
-                asgType     = src->gtType;
```

Comment:
When return type is TYP_STRUCT, but local variable is TYP_SIMD16, changing the local variable type is wrong.
Similar issue with asgType which will modify the type of the return.
Move inside GT_LCL_FLD to fix the issue.